analyzer/codechecker_analyzer/output_handler.py

```diff
@@ -0,0 +1,254 @@
+# -------------------------------------------------------------------------
+#                     The CodeChecker Infrastructure
+#   This file is distributed under the University of Illinois Open Source
+#   License. See LICENSE.TXT for details.
+# -------------------------------------------------------------------------
+"""
+Analyzer output formatter.
+"""
+from __future__ import print_function
+from __future__ import division
+from __future__ import absolute_import
+
+import hashlib
+import os
+import uuid
+
+from codechecker_common.logger import get_logger
+
+from statistics_collector.collectors.special_return_value import \
+    SpecialReturnValueCollector
+
+from . import analysis_manager, env
+from .analyzers.clangsa.analyzer import ClangSA
+from .analyzers.clangsa.ctu_manager import generate_ast_cmd, map_functions_cmd
+from .analyzers.clangsa.ctu_triple_arch import get_triple_arch
+from .analyzers.clangsa.statistics_manager import build_stat_coll_cmd
+
+
+LOG = get_logger('analyzer')
+
+
+class MakeFileCreator(object):
+    """ Creates a Makefile from analyzer actions. """
+
+    def __init__(self, analyzers, output_path, config_map, context,
+                 skip_handler, pre_analysis, statistics_data, ctu_data):
+        self.__analyzers = analyzers
+        self.__output_path = output_path
+        self.__config_map = config_map
+        self.__context = context
+        self.__skip_handler = skip_handler
+        self.__pre_analysis = pre_analysis
+
+        self.__ctu_data = ctu_data
+        self.__ctu_dir = None
+        self.__ctu_temp_fnmap_folder = None
+        if ctu_data:
+            self.__ctu_dir = ctu_data['ctu_dir'] if ctu_data else None
+            self.__ctu_temp_fnmap_folder = ctu_data['ctu_temp_fnmap_folder']
+
+        self.__statistics_data = statistics_data
+        self.__stat_tmp_dir = None
+        self.__stats_dir = None
+        if statistics_data:
+            self.__stat_tmp_dir = statistics_data['stat_tmp_dir']
+            self.__stats_dir = statistics_data['stats_out_dir']
+
+        self.__makefile = os.path.join(output_path, 'Makefile')
+
+        self.__analyzer_env = env.extend(context.path_env_extra,
+                                         context.ld_lib_path_extra)
+
+        self.__config = config_map.get(ClangSA.ANALYZER_NAME)
+
+        ctu_capability = config_map[ClangSA.ANALYZER_NAME].ctu_capability
+        self.__func_map_cmd = ctu_capability.mapping_tool_path
+
+    def __format_analyzer_type(self, analyzer_type):
+        """ Format the given analyzer type. """
+        return analyzer_type.replace('-', '')
+
+    def __get_target_name(self, action):
+        """ Get target name for the given action. """
+        analyzer_name = self.__format_analyzer_type(action.analyzer_type)
+        return hashlib.md5(analyzer_name + '_' + action.source).hexdigest()
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of insecure MD2, MD4, MD5, or SHA1 hash function.](https://app.codacy.com/app/CodeChecker/codechecker/pullRequest?prid=4606465)